src/main/java/org/jenkinsci/plugins/scriptsecurity/sandbox/whitelists/AclAwareWhitelist.java

```diff
@@ -50,6 +50,9 @@
      */
     public AclAwareWhitelist(Whitelist unrestricted, Whitelist restricted) {
         this.unrestricted = unrestricted;
+        if (this.unrestricted instanceof EnumeratingWhitelist) {
+            ((EnumeratingWhitelist) this.unrestricted).precache();
```

Comment:
OK, though as before it would be more modular to just have `EnumeratingWhitelist` set up its cache on its own the first time it is used.